stackstac/stack.py

```diff
@@ -276,6 +276,9 @@ def stack(
         The size of ``y`` and ``x`` will be determined by ``resolution`` and ``bounds``, which in many cases are
         automatically computed from the items you pass in.
     """
+    if rescale:
+        assert "float" in np.dtype(dtype).name, "The requested dtype is incompatible with rescale=True, which requires a floating-point dtype."
```

Comment:
```suggestion
        assert np.dtype(dtype).kind in ("f", "c"), "The requested dtype is incompatible with rescale=True, which requires a floating-point dtype."
```
nit: using [`dtype.kind`](https://numpy.org/doc/stable/reference/generated/numpy.dtype.kind.html) is probably the more proper way to write this test. Also, complex floats would be valid too (could come up for SAR data).

Reply:
Thanks! I was searching for a good way to do this and could not find one, this is handy.